Makefile

```diff
@@ -208,11 +208,22 @@ prepare: prepare-build
 prepare-help: prepare-build
 	mono --debug $(PREPARE_EXE) -h
 
+.PHONY: shutdown-compiler-server
+shutdown-compiler-server:
```

Comment:
A more "conventional" implementation would be:

```make
.PHONY: shutdown-compiler-server
shutdown-compiler-server:
	# Ensure the VBCSCompiler.exe process isn't running during the mono update
	pgrep -lfi VBCSCompiler.exe 2>/dev/null || true
	pid=`pgrep -lfi VBCSCompiler.exe 2>/dev/null | awk '{ print $$1 }'`;                    \
	echo "VBCSCompiler process ID (if running): $$pid" ;                                   \
	if [ -n "$$pid" ]; then                                                                 \
		echo "Terminating the VBCSCompiler '$$pid' server process prior to updating mono" ; \
		kill -HUP $$pid 2>/dev/null ;                                                       \
		echo "VBCSCompiler server process should no longer exist" ;                         \
		pgrep -lfi VBCSCompiler.exe 2>/dev/null || true ;                                   \
	fi
```

Reply:
updated to align on bash. indeed much cleaner than the Frankenstein approach of combining make and bash. commit: 8a3d64455a394724a94b78a28b9a035897ac8bfb